ec2api/api/security_group.py

```diff
@@ -247,7 +261,7 @@ def _build_rules(context, group_id, group_name, ip_permissions, direction):
             {'security_group_id': os_security_group_id,
              'direction': direction,
              'ethertype': 'IPv4'})
-        protocol = rule.get('ip_protocol', -1)
+        protocol = rule.get('ip_protocol', 'any')
```

Comment:
this code never hit if protocol is not belongs to [any tcp udp icmp] as in _validate param function call exception raised if protocol does not belong to any value /tcp/udp/icmp/any. so not sure why this  check is here. after going through the ec2 project it is clear that we need to clean most of the ec2 code.